official/resnet/resnet_run_loop.py

```diff
@@ -344,10 +345,14 @@ def resnet_main(flags, model_function, input_function):
           'multi_gpu': flags.multi_gpu,
           'version': flags.version,
       })
+  benchmark_logger = (logger.BenchmarkLogger(flags.benchmark_log_dir)
+                      if flags.benchmark_log_dir is not None else None)
 
   for _ in range(flags.train_epochs // flags.epochs_between_evals):
     train_hooks = hooks_helper.get_train_hooks(
-        flags.hooks, batch_size=flags.batch_size)
+        flags.hooks,
+        batch_size=flags.batch_size,
+        benchmark_log_dir=flags.benchmark_log_dir)
```

Comment:
Rather than require benchmark_log_dir, I think it would be convenient if it was:
`benchmark_log_dir if benchmark_log_dir is not None else model_dir`
as this makes manual runs with profiling easier.

Reply:
Will that result in unintended behavior? The logger should have to be turned on explicitly, and saving things to the model dir without explicit instruction might seem like an odd side effect.

Reply:
Specifically what I am suggesting is for the case when `benchmarklogginghook` is included but --benchmark_log_dir is not.

Reply:
In that case, it will spit out a error message when init the hook and ask user to config the flag.

Reply:
Yes. What I'm suggesting is that rather than spit out an error it uses model dir. That way in the case that you want the timeline with the rest of the info it saves a command line arg.

Reply:
Since the benchmarklogginghook is not there by default, when user actually specifying that in the hook list, I think its fair to ask them specifying where they want to log the benchmark.

Reply:
Alright. I'll leave the final say to you.